contract-tests/test_api.py

```diff
@@ -165,3 +165,106 @@ def test_recipe_history(conf, requests_session):
             created = datetime.strptime(revision["date_created"], "%Y-%m-%dT%H:%M:%S.%fZ")
             assert created < last_date
             last_date = created
+
+
+def test_docs(conf, requests_session):
+    r = requests_session.get(conf.getoption("server") + "/api/docs")
+    r.raise_for_status()
+    data = r.json()
+    assert data["_type"] == "document"
+    assert "_meta" in data
+
+
+def test_approval_request(conf, requests_session):
+    r = requests_session.get(conf.getoption("server") + "/api/v1/approval_request")
+    r.raise_for_status()
+    data = r.json()
+    sample_record = data[0]
+    expected_fields = ["approved", "comment", "created", "creator", "id"]
+    expected_approver_fields = ["email", "first_name", "id", "last_name"]
+    expected_creator_fields = ["email", "first_name", "id", "last_name"]
+    assert r.status_code in [200, 302]
+
+    for field in expected_fields:
+        assert field in sample_record
+
+    for field in expected_approver_fields:
+        assert field in sample_record["approver"]
+
+    for field in expected_creator_fields:
+        assert field in sample_record["creator"]
+
+
+def test_approval_request_by_record(conf, requests_session):
+    r = requests_session.get(conf.getoption("server") + "/api/v1/approval_request/1")
+    r.raise_for_status()
+    data = r.json()
+    expected_fields = ["approved", "approver", "comment", "created", "creator", "id"]
```

Comment:
This seems like a pretty shallow way of testing the response of the API. Wouldn't it be better to validate the response against a schema instead? I don't think this would have caught any problems we've seen recently that weren't caught by other tests.

Reply:
If there is a downloadable schema available, I'd be happy to use it and rework the tests to focus on validation

Reply:
We don't have any schemas already made. Simply checking for the presence keys doesn't seem like a very interesting test though.